vars/detectDockerAgent.groovy

```diff
@@ -6,21 +6,50 @@
  * this file be licensed under the Apache-2.0 license or a
  * compatible open source license.
  */
+
+/**@
+ * Contains logic to detect the Docker Agent and Images
+ *
+ * @param args A map of the following parameters
+ * @param args.manifest <optional> The input manifest path, default to "manifests/${INPUT_MANIFEST}"
+ * @param args.distribution <optional> The distribution name, default to "tar"
+ * @param args.platform <optional> The platform name, default to "linux"
+ */
 Map call(Map args = [:]) {
     def lib = library(identifier: "jenkins@main", retriever: legacySCM(scm))
     String manifest = args.manifest ?: "manifests/${INPUT_MANIFEST}"
+    String distribution = args.distribution ?: "tar"
+    String platform = args.platform ?: "linux"
+    echo "manifest: ${manifest}, dist: ${distribution}, plat: ${platform}"
+
     def inputManifest = lib.jenkins.InputManifest.new(readYaml(file: manifest))
-    dockerImage = inputManifest.ci?.image?.name ?: 'opensearchstaging/ci-runner:ci-runner-centos7-v1'
-    dockerArgs = inputManifest.ci?.image?.args
-    // Using default javaVersion as openjdk-17
-    String javaVersion = 'openjdk-17'
+    def schemaVersion = inputManifest.schemaVersion ?: "None"
+    echo "InputManifest SchemaVersion: ${schemaVersion}"
+
+    def dockerImage = 'opensearchstaging/ci-runner:ci-runner-al2-opensearch-build-v1'
+    def dockerArgs = '-e JAVA_HOME=/opt/java/openjdk-21'  // Using default javaVersion as openjdk-21
+    def javaVersion = 'openjdk-21'
+
+    if (schemaVersion < 1.2) {
```

Comment:
would this error out in case schema version is not provided and it ends up comparing 'None < 1.2`?

Reply:
I will just add a if (schemaVersion != 'None' && schemaVersion < 1.2)